api/grpcserver/post_service.go

```diff
@@ -52,11 +55,29 @@ func NewPostService(log *zap.Logger) *PostService {
 	}
 }
 
+// AllowConnections sets if the grpc service accepts new incoming connections from post services.
+func (s *PostService) AllowConnections(allow bool) {
+	s.clientMtx.Lock()
+	defer s.clientMtx.Unlock()
+	s.allowConnections = allow
+}
+
+// connectionAllowed returns if the grpc service accepts new incoming connections from post services.
+func (s *PostService) connectionAllowed() bool {
+	s.clientMtx.Lock()
+	defer s.clientMtx.Unlock()
+	return s.allowConnections
+}
+
 // Register is called by the PoST service to connect with the node.
 // It creates a bidirectional stream that is kept open until either side closes it.
 // The other functions on this service are called by services of the node to send
 // requests to the PoST node and receive responses.
 func (s *PostService) Register(stream pb.PostService_RegisterServer) error {
+	if !s.connectionAllowed() {
+		return status.Error(codes.PermissionDenied, "connection not allowed: node has no coinbase set in config")
```

Comment:
I'm not sure if `PERMISSION_DENIED` is the right code here. Quoting [GRPC docs](https://grpc.github.io/grpc/core/md_doc_statuscodes.html):
> The caller does not have permission to execute the specified operation. PERMISSION_DENIED must not be used for rejections caused by exhausting some resource (use RESOURCE_EXHAUSTED instead for those errors). PERMISSION_DENIED must not be used if the caller can not be identified (use UNAUTHENTICATED instead for those errors). This error code does not imply the request is valid or the requested entity exists or satisfies other pre-conditions. 


I think `FAILED_PRECONDITION` is a better match. 
> The operation was rejected because the system is not in a state required for the operation's execution. For example, the directory to be deleted is non-empty, an rmdir operation is applied to a non-directory, etc. Service implementors can use the following guidelines to decide between FAILED_PRECONDITION, ABORTED, and UNAVAILABLE: (a) Use UNAVAILABLE if the client can retry just the failing call. (b) Use ABORTED if the client should retry at a higher level (e.g., when a client-specified test-and-set fails, indicating the client should restart a read-modify-write sequence). (c) Use FAILED_PRECONDITION if the client should not retry until the system state has been explicitly fixed. E.g., if an "rmdir" fails because the directory is non-empty, FAILED_PRECONDITION should be returned since the client should not retry unless the files are deleted from the directory. 

It's not that the service has no permission to register. The node is in a wrong state and it needs to be fixed. Wdyt?